core/src/test/scala/org/apache/spark/scheduler/BlacklistTrackerSuite.scala

```diff
@@ -88,6 +88,86 @@ class BlacklistTrackerSuite extends SparkFunSuite with BeforeAndAfterEach with M
     new TaskSetBlacklist(conf, stageId, clock)
   }
 
+  def configureBlacklistAndScheduler(confs: (String, String)*): Unit = {
+    conf = new SparkConf().setAppName("test").setMaster("local")
+      .set(config.BLACKLIST_ENABLED.key, "true")
+    confs.foreach { case (k, v) => conf.set(k, v) }
+
+    clock.setTime(0)
+    listenerBusMock = mock[LiveListenerBus]
+    blacklist = new BlacklistTracker(listenerBusMock, conf, clock)
+  }
+
+  test("Blacklisting individual tasks and checking for SparkListenerEvents") {
+    configureBlacklistAndScheduler()
```

Comment:
I think the rest of the test code here has changed a lot since you first started working on this, but I'd avoid putting in a whole test case just for this.  Instead, I'd look at the existing test cases, find all occurrences of `blacklist.isNodeBlacklisted` and `blacklist.isExecutorBlacklisted`, and add appropriate `verify(listenerBusMock)` calls around all of those.